packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -124,6 +124,7 @@ export default class CashierStore extends BaseStore {
             const agent_error = ObjectUtils.getPropertyValue(await WS.p2pAgentInfo(), ['error', 'code']);
             if (agent_error !== 'PermissionDenied') {
                 this.setIsP2pVisible(true);
+                this.is_p2p_agent = !agent_error;
```

Comment:
let's set this outside the if condition

Reply:
we have to check if permissionDenied first, then decides whether its an agent?